tests/Doctrine/Tests/DBAL/Functional/DataAccessTest.php

```diff
@@ -432,6 +440,13 @@ public function testNativeArrayListSupport()
         $data = $stmt->fetchAll(PDO::FETCH_NUM);
         $this->assertEquals(5, count($data));
         $this->assertEquals(array(array(100), array(101), array(102), array(103), array(104)), $data);
+
+        $stmt = $this->_conn->executeQuery('SELECT test_int FROM fetch_table WHERE test_binary IN (?)',
+            array(array($binaryValue)), array(Connection::PARAM_STR_ARRAY));
```

Comment:
Looks like you will have to insert the data with types beforehand, otherwise the binary data won't be inserted properly:

```php
for ($i = 100; $i < 110; $i++) {
    $this->_conn->insert('fetch_table', array(
        'test_int'      => $i,
        'test_string'   => 'foo' . $i,
        'test_datetime' => '2010-01-01 10:10:10',
        'test_binary'   => $binaryValue
    ), array(
        'test_binary' => \PDO::PARAM_LOB
    ));
}
```